eqcorrscan/utils/plotting.py

```diff
@@ -1354,6 +1360,32 @@ def pretty_template_plot(template, background=False, picks=False, **kwargs):
     else:
         mintime = background.sort(['starttime'])[0].stats.starttime
     template.sort(['network', 'station', 'starttime'])
+    # Sort the template by epicentral distance
+    if event:
```

Comment:
Turns out I didn't see what you were doing here - I would prefer the method above than adding attributes to the trace, especially if you have to set a `999` value to allow it to happen.

Reply:
`utils.catalog_utils` certainly sounds like a good place to refactor this function to. I'm wondering about what exactly the sort-function should return thought: You suggest a list of ordered trace IDs, but is that the most useful / simple implementation?
The alternatives would be to return:
a) return trace ids for the sorted order
b) a sorted stream or 
c) return the indices of the traces for the sorted order, but don't sort in place. 

I'm thinking that because for
a) I can't see a really simple (1-line) way to arrange the traces in the returned order of trace_ids, 
b) wouldn't require any extra code in the executing function
c) a 1-liner would be enough to arrange the Stream from the returned indices. 

(Maybe this is just my incompetence in Python, but I'm wondering if returning the trace_ids makes it more complicated than it could be?)

Reply:
```python
sorted_stream = Stream([st.select(trace_id) for trace_id in sorted_trace_ids])
```
might be the one-liner of choice?

Indices would be simpler - go with that if you want!

Reply:
Ok, I went with indices. When using the new option "pick_time" to sort, there are some cases that are not obvious how they should best be handled - here's what happens now (see [test-image to compare behavior](https://github.com/flixha/EQcorrscan/blob/develop/eqcorrscan/tests/baseline/test_pretty_template_plot_sort_by_picktime.png)
- when there is more than one pick on a channel, then all appearances of that channel are sorted at the position of the earliest pick on that channel (this has an effect for templates where there is a P and S pick on the vertical)
- The different horizontal channels at one station are treated equally. That means even if a pick is only directly associated with one horizontal channel, then the other horizontal channel (`[NE12]`) will sort at an adjacent position.

If one, in contrast to sorting by pick_time as described here, just sort a template by the starttime of each template trace, then one can still just template.sort(["starttime"]).

Any comments on these choices? I made them because I think they make most sense in relation to plotting/sorting templates, but I'm aware that users may want to achieve other things with a function like this. I'm not sure one can find one set of "perfect" sort rules for this function, but of course it's nice if most users benefit.